### ISSUE TITLE
Tensorflow Serving not taking any inputs; returns logits as if it had no inputs; Nodejs

### ISSUE THREAD

**Issue by AfrazHussain** [tensorflow/serving#686]
I'm facing a issue in deploying my Tensorflow model in Nodejs. I have trained and tested a model for text categorization. I have set up everything, and the gRPC connections go through as well. I get an output from my Serving model but the output is just a standard output when there is no input. When I evaluate this model with a simple `feed_dict` in Python it works just fine. My model follows this model very closely: https://github.com/dennybritz/cnn-text-classification-tf/blob/master/text_cnn.py except now I have preprocessing within the Text CNN class.

Let me explain it in a bit detail (you can simply skip to the last part where I show my outputs if you want): As I mentioned, I have trained a text categorization model. I preprocess the models input within the graph as follows: 
```
# I load up my dictionary as follows:
vocab_processor = learn.preprocessing.VocabularyProcessor.restore(os.path.join(os.path.curdir, 'vocab'))
vocab_dict = list(vocab_processor.vocabulary_._mapping) # I then convert this to a TF lookup table
# ...
self.input_raw = tf.placeholder(tf.string, None, name="input_raw")
self.input_x = table.lookup(tf.string_split(self.input_raw))
# ... Followed by a few other transformations

# And I get the outputs as follows (again you can refer to the link for more detailed information)
self.scores = tf.nn.xw_plus_b(self.h_drop, W, b, name='scores')
self.probabilities = tf.nn.top_k(tf.nn.softmax(self.scores), 5, name='probabilities')
```

I then export this graph as follows:
```
saver = tf.train.import_meta_graph("{}.meta".format(checkpoint_file))
saver.restore(sess, checkpoint_file)

input_raw = graph.get_operation_by_name("input_raw").outputs[0]
probabilities, indices = graph.get_operation_by_name("output/probabilities").outputs
table = tf.contrib.lookup.index_to_string_table_from_tensor(tf.constant(labels))
prediction_classes = table.lookup(tf.to_int64(indices))

builder = tf.saved_model.builder.SavedModelBuilder(export_path)

inputs = tf.saved_model.utils.build_tensor_info(input_raw)
outputs_classes = tf.saved_model.utils.build_tensor_info(prediction_classes)
outputs_scores = tf.saved_model.utils.build_tensor_info(probabilities)

prediction_signature = (
    tf.saved_model.signature_def_utils.build_signature_def(
        inputs={
            "input_raw": inputs
        outputs={
            "classes": outputs_classes,
            "scores": outputs_scores},
        method_name=tf.saved_model.signature_constants.PREDICT_METHOD_NAME))

legacy_init_op = tf.group(tf.tables_initializer(), name='legacy_init_op')

builder.add_meta_graph_and_variables(
    sess, [tf.saved_model.tag_constants.SERVING],
    signature_def_map={
        tf.saved_model.signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY:
            prediction_signature,
    },
    legacy_init_op=legacy_init_op)

builder.save()
```

As mentioned earlier I deploy this in Nodejs, as follows:
```
const PROTO_PATH = "./pb/prediction_service.proto";
const TensorflowServing = grpc.load(PROTO_PATH).tensorflow.serving;

const client = new TensorflowServing.PredictionService(
    tf_localhost, grpc.credentials.createInsecure()
);

function getModelMsg(val){
    return {
        model_spec: { name: "model_export", signature_name: "serving_default", version: 1},
        inputs: {
            input_raw: {
                dtype: "DT_STRING",
                tensor_shape: {
                    dim: [{size: 1}],
                    unknown_rank: true
                },
                string_val: val
            }
        }
    }
}
# And I just pass on the sentence I want to classify through here.
```

This gives the following output in JSON: 
```
{"outputs":{"classes":{"dtype":"DT_STRING","tensor_shape:{"dim":[{"size":1,"name":""},{"size":5,"name":""}],"unknown_rank":false},"version_number":0,"tensor_content":{"type":"Buffer","data":[]},
 "string_val":[{"type":"Buffer","data":[77,111,110,101,121]} ... ]},
# I have removed some of the Buffer data, but I can verify that it is 5 classes from the lookup table that I provided in my model export
 "scores"{"dtype":"DT_FLOAT","tensor_shape":{"dim":[{"size":"1","name":""},{"size":"5","name":""}],"unknown_rank":false},"version_number":0,"tensor_content":{"type":"Buffer","data":[]},
 "float_val":[0.014690253883600235,0.00033853188157882,0.00853955152099133,0.008140762336552145,0.00756757420069599]}}}
```

As you can see, that the model seems to be connected and everything, but this is not the expected output. You only get this output when you evaluate the model with an empty string, or when the vocabulary doesn't find any words in the sentence (in which case it returns 0s), and thus this output.

I'm not sure what's going wrong here. Is there something wrong with inference in my export model? Is there something wrong with the inputs of my model? My model expects a vector to be fed into the `input_raw` placeholder. I have searched the issues here, but haven't found anything similar. Any help is appreciated.

**Comment by chrisolston** [comment]
Hmmm. I don't see anything obvious. Some things you could try are:
 1. Do the inference in python (without the model export, signature, RPC, etc.) and see if it works there.
 2. Inspect the request rpc to be sure the "string_val: val" part is populated with a non-empty string.
 3. Double-check the vocabulary.
 4. Add some printfs in the server code to see what tensor is getting fed in, and maybe isolate if/where it's becoming an empty string.

**Comment by sukritiramesh** [comment]
To add to the suggestions, you can also do one-off inspections and executions of SavedModels using the SavedModel CLI to check the state of your export: https://www.tensorflow.org/versions/master/programmers_guide/saved_model#cli_to_inspect_and_execute_savedmodel

**Comment by gautamvasudevan** [comment]
Closing due to staleness.
